### PR TITLE
Add option to skip keys with the value of `null` in `.stringify()`

### LINKED ISSUE #196: `skipNulls` option for queryString.stringify
Hi!

I am now comparing 2 libs for future use on my project: `qs` and `query-string`.
The second one has some advantages like encoding/decoding booleans and numbers which are important for me. But the 1st one has `skipNulls` option which allows to skip `null` and `undefined` values:

```js

const obj1 = {
  a: 1,
  b: '2',
  c: [3, 4, 5],
  d: true,
  e: 'just string',
  f: null,
  g: undefined,
};

// qs
// can skip null/undefined values
const ex2 = qs.stringify(obj1, {
  addQueryPrefix: true,
  arrayFormat: 'comma',
  skipNulls: true,
});
const ex3 = qs.parse(ex2, {
  parseArrays: true,
  comma: true,
  ignoreQueryPrefix: true,
});

// query-string
// can store numbers and booleans
const ex4 = queryString.stringify(obj1, {
  strict: true,
});
const ex5 = queryString.parse(ex4, {
  parseNumbers: true,
  parseBooleans: true,
});

console.log('ex2', ex2);
console.log('ex3', ex3);
console.log('ex4', ex4);
console.log('ex5', ex5);
```

Screenshot from 2019-07-26 10-48-56

### Does it sound reasonable to implement this option?

### PR DESCRIPTION
**Pull request by yaodingyd**
Follow up on #197 to fix #196.

Closes #197

### CODE REVIEW

**Comment by yaodingyd** [comment]
Additionally, fix a bug in current stringify behavior: if `arrayFormat` is set to `comma` and first item is null, the result won't have the key in it. something like
```javascript
queryString.stringify({a: [null, 1, undefined, 2, null]}, {
		arrayFormat: 'comma'
	}
// => `,1,2`, missing `a=`
```

**Comment by sindresorhus** [comment]
What about https://github.com/sindresorhus/query-string/issues/196#issuecomment-522169947 ?

File: index.js [hunk 1]
```diff
@@ -257,7 +257,17 @@ exports.stringify = (object, options) => {
 	}, options);
 
 	const formatter = encoderForArrayFormat(options);
-	const keys = Object.keys(object);
+
+	const objectCpy = Object.assign({}, object);
```
Comment (sindresorhus) [inline index.js]:
```suggestion
	const objectCopy = Object.assign({}, object);
```

File: readme.md [hunk 2]
```diff
@@ -204,10 +204,33 @@ queryString.stringify({b: 1, c: 2, a: 3}, {sort: false});
 
 If omitted, keys are sorted using `Array#sort()`, which means, converting them to strings and comparing strings in Unicode code point order.
 
+##### skipNullAndUndefined
+
+Skip the key in stringify result if value is `null` or `undefined`.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
Skip the element in the stringify result if the value is `null` or `undefined`.
```

File: readme.md [hunk 3]
```diff
@@ -204,10 +204,33 @@ queryString.stringify({b: 1, c: 2, a: 3}, {sort: false});
...
+	skipNullAndUndefined: true,
+});
+//=> ''
+```
+
 ### .extract(string)
 
 Extract a query string from a URL that can be passed into `.parse()`.
 
+Note: This behaviour can be changed with the `skipNullAndUndefined` option.
```
Comment (sindresorhus) [inline readme.md]: This should be added to index.d.ts too.

**Comment by yaodingyd** [comment]
which direction are we going? the option should be called `skipNull` or change the current behavior so that `undefined` is not skipped?

**Comment by sindresorhus** [comment]
> the option should be called skipNull 

This

File: index.d.ts [hunk 4]
```diff
@@ -192,6 +192,26 @@ export interface StringifyOptions {
 	```
 	*/
 	readonly sort?: ((itemLeft: string, itemRight: string) => number) | false;
+
+	/**
+	Skip the key in stringify result if value is `null` or `undefined`.
```
Comment (sindresorhus) [inline index.d.ts]: This makes it sound like it will **not** skip `undefined` by default, which it does.

**Pull request merged**